core/trino-main/src/main/java/io/trino/execution/DropTableTask.java

```diff
@@ -76,7 +77,14 @@ public ListenableFuture<Void> execute(
                     "Table '%s' does not exist, but a view with that name exists. Did you mean DROP VIEW %s?", originalTableName, originalTableName);
         }
 
-        RedirectionAwareTableHandle redirectionAwareTableHandle = metadata.getRedirectionAwareTableHandle(session, originalTableName);
+        RedirectionAwareTableHandle redirectionAwareTableHandle;
+        try {
+            redirectionAwareTableHandle = metadata.getRedirectionAwareTableHandle(session, originalTableName);
+        }
+        catch (TrinoException e) {
```

Comment:
why is this exception-driven?
what potential exceptions are we ignoring here?

Reply:
I want to check the exception which fails with the `METADATA_NOT_FOUND` error.

Reply:
We should probably rethink here the way we obtain the `tableName`. We don't need actually the full `tableHandle` for dropping the table, right?

Reply:
> I want to check the exception which fails with the `METADATA_NOT_FOUND` error.

 `metadata.dropTable` must support such case anyway
what does it currently throw?